api/src/main/java/org/pmiops/workbench/google/DirectoryServiceImpl.java

```diff
@@ -113,20 +118,42 @@ public boolean isUsernameTaken(String username) {
   public User createUser(String givenName, String familyName, String username, String contactEmail) {
     String gSuiteDomain = configProvider.get().googleDirectoryService.gSuiteDomain;
     String password = randomString();
+    Alias alias = new Alias();
+    alias.setAlias(contactEmail);
     User user = new User()
       .setPrimaryEmail(username+"@"+gSuiteDomain)
       .setPassword(password)
       .setName(new UserName().setGivenName(givenName).setFamilyName(familyName))
-      .setEmails(new UserEmail().setType("custom").setAddress(contactEmail).setCustomType("contact"))
+//      .setAliases(Arrays.asList(contactEmail))
+//      .setEmails(new UserEmail().setAddress(contactEmail).setType("custom").setCustomType("contact").setPrimary(false))
       .setChangePasswordAtNextLogin(true);
+    log.log(Level.INFO, "user before insert: " + user.toString());
     retryHandler.run((context) -> getGoogleDirectoryService().users().insert(user).execute());
+    retryHandler.run((context) -> getGoogleDirectoryService().users().aliases().insert(username+'@'+gSuiteDomain, alias).execute());
     try {
       mailServiceProvider.get().sendWelcomeEmail(contactEmail, password, user);
     } catch (MessagingException e) {
       throw new WorkbenchException(e);
     }
     return user;
   }
+
+  @Override
+  public void resendWelcomeEmail(String userKey) {
+    User user = getUser(userKey);
+    String password = randomString();
+    user.setPassword(password);
+//    retryHandler.run((context) -> getGoogleDirectoryService().users().update(userKey, user).execute());
+    Aliases emails = retryHandler.run((context) -> getGoogleDirectoryService().users().aliases().list(userKey).execute());
+    log.log(Level.INFO, "user: " + user.toString());
+    log.log(Level.INFO, "emails: " + emails.getAliases().toString());
```

Comment:
This is clearly still in the "figuring it out stage". The commented line above here does need to be uncommented once everything else is figured out.